service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/JsonObjectMapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2015 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.poseidon.serviceclients;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Created by venkata.lakshmi on 16/04/15.
+ *
+ *
+ */
+public abstract class JsonObjectMapper {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public JsonObjectMapper() {
+        configure(objectMapper);
+    }
+    public abstract void configure(final ObjectMapper objectMapper);
+
+    public ObjectMapper getObjectMapper() {
```

Comment:
Mark this as final. We don't want implementing classes to override this and return a new object mapper